### PR TITLE
[SPARK-16299][SPARKR] Capture errors from R workers in daemon.R to avoid deletion of R session temporary directory.

### PR DESCRIPTION
**Pull request by sun-rui**
## What changes were proposed in this pull request?

Capture errors from R workers in daemon.R to avoid deletion of R session temporary directory. See detailed description at https://issues.apache.org/jira/browse/SPARK-16299
## How was this patch tested?

SparkR unit tests.

### CODE REVIEW

**Comment by sun-rui** [comment]
@shivaram, @davies

**Comment by shivaram** [comment]
@sun-rui This is a great find ! I will test this locally as well

cc @JoshRosen who was running into the same problem with pipeRDD tests

File: R/pkg/inst/worker/daemon.R [hunk 1]
```diff
@@ -44,7 +44,7 @@ while (TRUE) {
     if (inherits(p, "masterProcess")) {
       close(inputCon)
       Sys.setenv(SPARKR_WORKER_PORT = port)
-      source(script)
+      try(source(script))
```
Comment (felixcheung) [inline R/pkg/inst/worker/daemon.R]: is it worthwhile to do tryCatch and in the error handler, print the error?

Comment (sun-rui) [inline R/pkg/inst/worker/daemon.R]: error still to stderr with try()

**Comment by shivaram** [comment]
Thanks @sun-rui -- I just tested this and this fixes the issue with the `pipeRDD` unit test on my machine.
LGTM. Merging this to master, branch-2.0

**Pull request closed without merging**